src/main/resources/data/spectrum/advancements/lategame/enchant_exchanging_staff.json

```diff
@@ -20,7 +20,7 @@
     }
   },
   "criteria": {
-    "enchanted_staff": {
+    "enchanted_item": {
```

Comment:
wait, what? Why? o.O

Reply:
Still why o.O
I used the Enchanter to enchant an Exchanging Staff with Silk Touch and it unlocked the advancement just fine.

Reply:
I swear it just did not work

Reply:
I can reproduce it not working in AoF7, for whatever reason. It works fine in dev.
Changing the free text criterion name won't help you though, you'll just make players lose their progress. :P

Reply:
oh yeah >w< idk no idea, some vanilla bug?

Reply:
Good question. I'll look into it further when I get the time. May also be a mod compat thing.